### PR TITLE
Auth/ps 2298 reorg auth

### PR DESCRIPTION
**Pull request by MGibson1**
## Type of change

```
- [ ] Bug fix
- [ ] New feature development
- [ ] Tech debt (refactoring, code cleanup, dependency upgrades, etc)
- [ ] Build/deploy pipeline (DevOps)
- [x] Other
```

## Objective

Establish Auth team code owners

## Before you submit

- Please add **unit tests** where it makes sense to do so (encouraged but not required)
- If this change requires a **documentation update** - notify the documentation team
- If this change has particular **deployment requirements** - notify the DevOps team

### CODE REVIEW

File: apps/browser/src/auth/popup/services/index.ts [hunk 1]
```diff
@@ -0,0 +1,2 @@
+export { LockGuardService } from "./lock-guard.service";
```
Comment (MGibson1) [inline apps/browser/src/auth/popup/services/index.ts:1]: This is actually a net-new file. Let me know if you're uncomfortable with this level of code change and I can move back to a non-indexed services

**Comment by MGibson1** [comment]
This should be good to go. There will be a follow up that I need to merge rather than squash for one file not detected as a move due to its short length

**Review by differsthecat — COMMENTED** [review]
@MGibson1 Did you pull in latest from master? It appears you have a number of conflicts (from me most likely :cone-of-shame:)
There also appear to be some files that can be renamed or removed from whitelist.txt based on the lint error.

File: apps/browser/src/auth/scripts/duo.js [hunk 2]
```diff
@@ -0,0 +1,420 @@
+/**
```
Comment (differsthecat) [inline apps/browser/src/auth/scripts/duo.js:1]: Looks like it picked this file move as an add and delete

Comment (MGibson1) [reply apps/browser/src/auth/scripts/duo.js]: yes, I had forgotten to exclude from prettier. Now fixed, ty

**Comment by MGibson1** [comment]
yup, was working on the merge conflicts while I had this up, perhaps I should have done that first -- sorry.

I believe it's good to go now

File: apps/browser/src/auth/popup/services/index.ts (hunk 1 above)
Comment (differsthecat) [reply apps/browser/src/auth/popup/services/index.ts:1]: I'm fine with it!
What exactly does this do? I've seen this before but don't understand the benefit.

File: apps/browser/src/background/main.background.ts [hunk 3]
```diff
@@ -30,21 +28,28 @@ import {
...
 import { SystemService as SystemServiceAbstraction } from "@bitwarden/common/abstractions/system.service";
-import { TokenService as TokenServiceAbstraction } from "@bitwarden/common/abstractions/token.service";
 import { TotpService as TotpServiceAbstraction } from "@bitwarden/common/abstractions/totp.service";
-import { TwoFactorService as TwoFactorServiceAbstraction } from "@bitwarden/common/abstractions/twoFactor.service";
 import { UserVerificationApiServiceAbstraction } from "@bitwarden/common/abstractions/userVerification/userVerification-api.service.abstraction";
 import { UserVerificationService as UserVerificationServiceAbstraction } from "@bitwarden/common/abstractions/userVerification/userVerification.service.abstraction";
 import { UsernameGenerationService as UsernameGenerationServiceAbstraction } from "@bitwarden/common/abstractions/usernameGeneration.service";
 import { VaultTimeoutService as VaultTimeoutServiceAbstraction } from "@bitwarden/common/abstractions/vaultTimeout/vaultTimeout.service";
 import { VaultTimeoutSettingsService as VaultTimeoutSettingsServiceAbstraction } from "@bitwarden/common/abstractions/vaultTimeout/vaultTimeoutSettings.service";
```
Comment (differsthecat) [inline apps/browser/src/background/main.background.ts:36]: Should VaultTimeout go to Auth as well?

Comment (MGibson1) [reply apps/browser/src/background/main.background.ts:36]: I think the answer is yes, but I made a ticket for it (https://bitwarden.atlassian.net/browse/PM-29). I wasn't sure at the time, but I can't think of another team to own it. Figured it'd fall out in the shuffle.

File: apps/browser/src/auth/popup/services/index.ts (hunk 1 above)
Comment (MGibson1) [reply apps/browser/src/auth/popup/services/index.ts:1]: index files (barrel files) are a relatively common pattern for establishing a folder as a module. Think `__init__.py` or `mod.rs`. In all cases, they take exports from files in the same directory, and elevate them as _directory_ exports.

This lets you make mini-packages in individual directories where exported content is not necessarily accessible from the outside. By doing this, we could have a base guard class in this directory that is imported by both `lock` and `unauth` guards, but not exported beyond the folder.

In TS, we would need to enable linter rules to disallow importing past an index file, but that's planned work.

I'd be OK deleting this largely because it's not actually very useful, it just reduces the number of import lines required when importing our guard services.

Comment (differsthecat) [reply apps/browser/src/auth/popup/services/index.ts:1]: Got it, thank you!
I'm cool with leaving it, Auth can remove it if they feel it's not needed :p

File: apps/desktop/src/auth/hint.component.ts [hunk 4]
```diff
@@ -15,9 +15,9 @@ import { PlatformUtilsService } from "@bitwarden/common/abstractions/platformUti
 export class HintComponent extends BaseHintComponent {
   constructor(
     router: Router,
+    platformUtilsService: PlatformUtilsService,
     i18nService: I18nService,
     apiService: ApiService,
-    platformUtilsService: PlatformUtilsService,
```
Comment (differsthecat) [inline apps/desktop/src/auth/hint.component.ts:20]: Any specific reason for this change? Seems out of scope for this

Comment (MGibson1) [reply apps/desktop/src/auth/hint.component.ts:20]: looks like a bad merge. I'll revert

**Review by differsthecat — APPROVED** [review]
Looks good!
